etna/packages/etna-js/components/__tests__/TreeView.spec.jsx

```diff
@@ -0,0 +1,193 @@
+            .simulate('change');
+          expect(test.selected).toEqual([
+            ['Top Level 1', 'Inner 2', 'Inner Inner 1'],
+            ['Top Level 2']
+          ]);
+          expect(test.lastSelectionsByCallback).toEqual([
+            'Inner Inner 1',
+            'Top Level 2'
+          ]);
+
+          test
+            .findInput(['Top Level 1', 'Inner 2'], 'Inner Inner 2')
+            .simulate('change');
+          expect(test.selected).toEqual([
+            ['Top Level 1', 'Inner 2', 'Inner Inner 1'],
+            ['Top Level 1', 'Inner 2', 'Inner Inner 2'],
+            ['Top Level 2']
+          ]);
+          expect(test.lastSelectionsByCallback).toEqual([
+            'Inner Inner 1',
+            'Inner Inner 2',
+            'Top Level 2'
+          ]);
+
+          test.findInput(['Top Level 1'], 'Inner 1').simulate('change');
+          test.findInput(['Top Level 1'], 'Inner 3').simulate('change');
+          expect(test.selected).toEqual([
+            ['Top Level 1', 'Inner 1'],
+            ['Top Level 1', 'Inner 2', 'Inner Inner 1'],
+            ['Top Level 1', 'Inner 2', 'Inner Inner 2'],
+            ['Top Level 1', 'Inner 3'],
+            ['Top Level 2']
+          ]);
+
+          expect(test.lastSelectionsByCallback).toEqual([
+            'Inner 1',
+            'Inner Inner 1',
+            'Inner Inner 2',
+            'Inner 3',
+            'Top Level 2'
+          ]);
+
+          test
+            .findInput(['Top Level 1', 'Inner 2'], 'Inner Inner 3')
+            .simulate('change');
+          expect(test.selected).toEqual([
+            ['Top Level 1'],
+            ['Top Level 1', 'Inner 1'],
+            ['Top Level 1', 'Inner 2'],
+            ['Top Level 1', 'Inner 2', 'Inner Inner 1'],
+            ['Top Level 1', 'Inner 2', 'Inner Inner 2'],
+            ['Top Level 1', 'Inner 2', 'Inner Inner 3'],
+            ['Top Level 1', 'Inner 3'],
+            ['Top Level 2']
+          ]);
+          expect(test.lastSelectionsByCallback).toEqual([
+            'Inner 1',
+            'Inner Inner 1',
+            'Inner Inner 2',
+            'Inner Inner 3',
+            'Inner 3',
+            'Top Level 2'
+          ]);
+        });
+      });
+
+      describe('and then re-selecting that top level', () => {
+        beforeEach(() => {
+          test.findInput([], 'Top Level 1').simulate('change');
+        });
+
+        it('enables recursively child nodes', () => {
+          expect(test.selected).toEqual([
+            ['Top Level 1'],
+            ['Top Level 1', 'Inner 1'],
+            ['Top Level 1', 'Inner 2'],
+            ['Top Level 1', 'Inner 2', 'Inner Inner 1'],
+            ['Top Level 1', 'Inner 2', 'Inner Inner 2'],
+            ['Top Level 1', 'Inner 2', 'Inner Inner 3'],
+            ['Top Level 1', 'Inner 3'],
+            ['Top Level 2']
+          ]);
+
+          expect(test.lastSelectionsByCallback).toEqual([
+            'Inner 1',
```

Comment:
This test was failing, so I added this node because it seems like a leaf and should be included in the results from `lastSelectionsByCallback`?